app/helpers/comments_helper.rb

```diff
@@ -269,6 +284,19 @@ def do_cancel_delete_comment_link(comment)
     end
   end
 
+  def freeze_comment_button(comment)
+    if comment.on_ice?
+      button_to ts("Unfreeze"), freeze_comment_path(comment), method: :put
```

Comment:
Would it make sense to have Freeze/Unfreeze as two separate actions? As is, if two people are trying to moderate at the same time (e.g. admins on an admin post), one of them might click the "Freeze" button on a comment that the other has already frozen, and end up accidentally unfreezing the comment instead.